src/pool/srv_pool_check.c

```diff
@@ -84,15 +84,16 @@ pool_glance(uuid_t uuid, char *path, struct ds_pool_clue *clue_out)
 	}
 
 	rc = ds_pool_svc_load(&tx, uuid, &root, &global_version, &map_buf, &clue.psc_map_version);
-	if (rc == DER_UNINIT) {
+	if (rc == 0) {
+		D_FREE(map_buf);
+	} else if (rc == DER_UNINIT) {
```

Comment:
Just want to know why ds_pool_svc_load() return positive number for empty pool map instead of "-DER_UNINIT"?

Reply:
Because that is a important, non-error scenario that will almost 100% happen during the creation of a PS, and I was afraid that functions called by ds_pool_svc_load may return `-DER_UNINIT` for other unrelated scenarios. Hence, I chose a positive return value many years ago.

If I were to make the same choice again today, I might prefer defining an `enum` with a positive value.

Reply:
> Because that is a important, non-error scenario that will almost 100% happen during the creation of a PS, and I was afraid that functions called by ds_pool_svc_load may return `-DER_UNINIT` for other unrelated scenarios. Hence, I chose a positive return value many years ago.
> 
> If I were to make the same choice again today, I might prefer defining an `enum` with a positive value.

I think you can do that in master independently.